src/main/java/gift/oauth/OauthService.java

```diff
@@ -40,19 +39,11 @@ public KakaoToken getKakaoToken(String code){
             .retrieve()
             .toEntity(String.class);
 
-
         return extractKakaoToken(response.getBody());
     }
 
-    public KakaoToken extractKakaoToken(String responseBody){
-        JsonObject jsonObject = JsonParser.parseString(responseBody).getAsJsonObject();
-        String accessToken = jsonObject.get("access_token").getAsString();
-        String tokenType = jsonObject.get("token_type").getAsString();
-        String refreshToken = jsonObject.get("refresh_token").getAsString();
-        int expiresIn = jsonObject.get("expires_in").getAsInt();
-        String scope = jsonObject.get("scope").getAsString();
-        int refreshTokenExpiresIn = jsonObject.get("refresh_token_expires_in").getAsInt();
-
-        return new KakaoToken(accessToken,tokenType,refreshToken,expiresIn,scope,refreshTokenExpiresIn);
+    public KakaoToken extractKakaoToken(String responseBody) {
+        Gson gson = new Gson();
```

Comment:
ObjectMapper 역시 스프링이 등록해둔 bean이 있어서, 의존관계 주입을 받아서 사용하면 됩니다!
참고로 ObjectMapper는 정말 무거운 객체라 매번 만들고 GC되고 하는 것이 상당히 비효율적입니다ㅎㅎ
```suggestion
        return objectMapper.readValue(responseBody, KakaoToken.class);
```